src/ShelfLayoutManager.Infrastructure/CabinetsDataService.cs

```diff
@@ -0,0 +1,116 @@
+using FluentResults;
+using Microsoft.EntityFrameworkCore;
+using ShelfLayoutManager.Core;
+using ShelfLayoutManager.Infrastructure.Converters;
+
+namespace ShelfLayoutManager.Infrastructure;
+
+/// <inheritdoc>
+public class CabinetsDataService(DbContext dbContext, ICabinetEntityConverter cabinetEntityConverter)
+    : ICabinetsDataService
+{
+    private readonly DbContext _dbContext = dbContext;
+
+    private readonly ICabinetEntityConverter _cabinetEntityConverter = cabinetEntityConverter;
+
+    public async Task<Result<Cabinet?>> GetAsync(long id, bool includeRows, bool includeLanes, CancellationToken ct) =>
+        _cabinetEntityConverter.ConvertToCabinet(
+            await GetReadQueryable(includeRows, includeLanes).Where(c => c.Id == id).FirstOrDefaultAsync(ct));
+
+    public async Task<Result<IList<Cabinet>>> ListAsync(bool includeRows, bool includeLanes, CancellationToken ct) =>
+        (await GetReadQueryable(includeRows, includeLanes).ToListAsync(ct))
+            .Select(_cabinetEntityConverter.ConvertToCabinet).OfType<Cabinet>().ToArray();
+
+    public async Task<Result> AssignSkuAsync(CabinetSkuAssignment cabinetSkuAssignment, CancellationToken ct)
+    {
+        // TODO(abalkar): Add validation for given assignment.
+
+        bool getFromLane = cabinetSkuAssignment.FromRowNumber != 0 || cabinetSkuAssignment.FromLaneNumber != 0;
+        bool getToLane = cabinetSkuAssignment.ToRowNumber != 0 || cabinetSkuAssignment.ToLaneNumber != 0;
```

Comment:
Seems like zero is the sentinel value. Shouldn't it be an `&&` and not `||`, then?